dandiapi/api/tests/factories.py

```diff
@@ -19,58 +18,55 @@
     UserMetadata,
     Version,
 )
+from dandiapi.api.services.dandiset import star_dandiset
 from dandiapi.api.services.permissions.dandiset import add_dandiset_owner
 
 
-class UserMetadataFactory(factory.django.DjangoModelFactory):
-    class Meta:
-        model = UserMetadata
-
-    status = UserMetadata.Status.APPROVED.value
-
-
 class UserFactory(factory.django.DjangoModelFactory):
     class Meta:
         model = User
         skip_postgeneration_save = True
 
-    username = factory.SelfAttribute('email')
-    email = factory.Faker('free_email')
+    username = factory.Faker('user_name')
+    email = factory.Faker('email', safe=False)
     first_name = factory.Faker('first_name')
     last_name = factory.Faker('last_name')
 
-    metadata = factory.RelatedFactory(UserMetadataFactory, factory_related_name='user')
+    metadata = factory.RelatedFactory(
+        'dandiapi.api.tests.factories.UserMetadataFactory', factory_related_name='user'
+    )
+    social_account = factory.RelatedFactory(
+        'dandiapi.api.tests.factories.SocialAccountFactory', factory_related_name='user'
+    )
+
+
+class UserMetadataFactory(factory.django.DjangoModelFactory):
+    class Meta:
+        model = UserMetadata
+
+    user = factory.SubFactory(UserFactory, metadata=None)
+    status = UserMetadata.Status.APPROVED.value
 
 
 class SocialAccountFactory(factory.django.DjangoModelFactory):
     class Meta:
         model = SocialAccount
 
-    user = factory.SubFactory(UserFactory)
+    user = factory.SubFactory(UserFactory, social_account=None)
     uid = factory.Faker('sha1')
 
-    @factory.lazy_attribute
-    def extra_data(self) -> dict[str, Any]:
-        first_name = self.user.first_name
-        last_name = self.user.last_name
-        name = f'{first_name} {last_name}'
-
-        # Supply a fake created date at least 1 year before now
-        created = (
-            faker.Faker()
-            .date_time_between(
-                end_date=datetime.datetime.now(datetime.UTC) - datetime.timedelta(days=365)
-            )
-            .isoformat()
-        )
-
-        # Supply different values from User object, since social account values maybe be different
-        return {
-            'login': faker.Faker().user_name(),
-            'name': name,
-            'email': faker.Faker().ascii_email(),
-            'created_at': created,
+    extra_data = factory.Dict(
+        {
+            'name': factory.LazyAttribute(
+                lambda self: (
+                    f'{self.factory_parent.user.first_name} {self.factory_parent.user.last_name}'
+                )
+            ),
+            'login': factory.SelfAttribute('..user.username'),
+            'email': factory.SelfAttribute('..user.email'),
```

Comment:
Note, this is a change in policy. Before, a code comment indicated that the `SocialAccount` intentionally set different values (e.g. for `'email'`) than the `User`, since these could differ in the database.

However, since all users should be now created exclusively via social login, I think it makes more sense for these to just match. This makes it easier to access the authoritative values directly from the user object.

If this is reasonable, I might want to do a little more follow-up to ensure this change is respected everywhere. If we should expect that the `SocialAccount` and `User` have different metadata, than I'll also need to make some updates.

Reply:
We currently set the `username` field to the email supplied from the social account. So in reality, due to the github login flow, the username is almost always set to the `email` value from `SocialAccount`. 

It it sometimes (rarely) an even different value, due to differences in how people can specify emails on Github, changing github profile info after first login, etc.

While that's something that ought to be fixed in its own right, I think for now our tests would be more robust if we didn't make that assumption.